synapse/handlers/register.py

```diff
@@ -845,6 +854,9 @@ class and RegisterDeviceReplicationServlet.
         # This can only run on the main process.
         assert isinstance(self.device_handler, DeviceHandler)
 
+        # Prune the user's device list if they already have a lot of devices.
+        await self._prune_too_many_devices(user_id)
```

Comment:
Maybe this is clearer?

```suggestion
        # Prune the user's device list if they already have a lot of devices.
        await self._maybe_prune_too_many_devices(user_id)
```